### DISCUSSION TITLE
Keycloak configuration - encrypted passwords

### DISCUSSION THREAD

**Discussion by Satishshelake** [keycloak/keycloak#20840] [Archive]
Hello,
   I am looking for database password encryption in keycloak.conf file or vault implementation.
I have tried custom vault implementation but it does not support in keycloak.conf db-password property. 
How we can avoid plain text of passwords in keycloak.conf file?
Could anybody help in it?

**Comment by vmuzikar** [comment]
Hello, please see the following PR which will be included in the upcoming Keycloak 22.
https://github.com/keycloak/keycloak/pull/20375

**Reply by Satishshelake** [reply]
Thank you for your reply.. now waiting for new release

**Reply by augustinemmel** [reply]
Hi ,  Can you update on the status of https://github.com/keycloak/keycloak/pull/20375

I am looking for database password with vault implementation.

**Reply by vmuzikar** [reply]
It was part of the 22.0.0 release.

**Reply by augustinemmel** [reply]
can you provide the documentation link  for retrieve keycloak db password from vault . looking to integrate azure key vault to get the db password for keycloak at run time

**Reply by Satishshelake** [reply]
It doesn't support trustore encrypted password. any solution for same?

**Reply by vmuzikar** [reply]
@augustinemmel @Satishshelake Please see: https://www.keycloak.org/server/configuration#_setting_sensitive_options_using_a_java_keystore_file

**Reply by augustinemmel** [reply]
i am expecting to fetch from  external key vaults like azure key vault or hashicorp vault

**Reply by vmuzikar** [reply]
@augustinemmel Feel free to open an RFE. :)

**Reply by ncphins** [reply]
> @augustinemmel @Satishshelake Please see: https://www.keycloak.org/server/configuration#_setting_sensitive_options_using_a_java_keystore_file

Sorry to resurrect an old thread, but we have the same issue. I have moved the database password into a Java Keystore (as suggested) but that seems to require a clear password in the conf file for the Java Keystore? Our IT team is concerned with the clear password there. Am I missing something? 

```
config-keystore=keystore.p12
config-keystore-password=clear-password
```

**Reply by vmuzikar** [reply]
As mentioned above, we do not support fetching the keystore password from external vaults at the moment. However, the password can be specified via various [config sources](https://www.keycloak.org/server/configuration#_configuring_sources_for_keycloak), e.g. env vars which can be filled from an external vault.

**Reply by ncphins** [reply]
Thanks for the clarification. That was my understanding but I wanted to make sure before I went back to my team.
